src/resultset/formatters/resultset_replyverbose.c

```diff
@@ -217,33 +218,43 @@ static void _ResultSet_VerboseReplyWithVector
 
 void ResultSet_EmitVerboseRow
 (
-	RedisModuleCtx *ctx,
-	bolt_client_t *bolt_client,
-	GraphContext *gc,
-	SIValue **row,
-	uint numcols
+	ResultSet *set,
+	SIValue **row
 ) {
+	RedisModuleCtx *ctx = set->ctx;
 	// Prepare return array sized to the number of RETURN entities
-	RedisModule_ReplyWithArray(ctx, numcols);
+	RedisModule_ReplyWithArray(ctx, set->column_count);
 
-	for(int i = 0; i < numcols; i++) {
+	for(int i = 0; i < set->column_count; i++) {
 		SIValue v = *row[i];
-		_ResultSet_VerboseReplyWithSIValue(ctx, gc, v);
+		_ResultSet_VerboseReplyWithSIValue(ctx, set->gc, v);
 	}
 }
 
 // Emit the alias or descriptor for each column in the header.
 void ResultSet_ReplyWithVerboseHeader
 (
-	RedisModuleCtx *ctx,
-	bolt_client_t *bolt_client,
-	const char **columns,
-	uint *col_rec_map
+	ResultSet *set
 ) {
-	uint columns_len = array_len(columns);
-	RedisModule_ReplyWithArray(ctx, columns_len);
-	for(uint i = 0; i < columns_len; i++) {
+	RedisModuleCtx *ctx = set->ctx;
+	if(set->column_count > 0) {
+		// prepare a response containing a header, records, and statistics
+		RedisModule_ReplyWithArray(ctx, 3);
+	} else {
+		// prepare a response containing only statistics
+		RedisModule_ReplyWithArray(ctx, 1);
+		return;
+	}
+	RedisModule_ReplyWithArray(ctx, set->column_count);
+	for(uint i = 0; i < set->column_count; i++) {
 		// Emit the identifier string associated with the column
-		RedisModule_ReplyWithStringBuffer(ctx, columns[i], strlen(columns[i]));
+		RedisModule_ReplyWithStringBuffer(ctx, set->columns[i], strlen(set->columns[i]));
 	}
 }
+
+void ResultSet_EmitVerboseStats
+(
+	ResultSet *set
+) {
+	ResultSetStat_emit(set->ctx, &set->stats);
```

Comment:
This is strange, to have a formatter call the result-set in order to emit data,
I understand this was done probably for code reuse, if that's indeed the case we need to find a better place for `ResultSetStat_emit` logic to reside at.